src/commands/router.ts

```diff
@@ -88,7 +88,15 @@ export async function onInteractionReceived(interaction: Interaction) {
 
     await interaction.deferReply();
 
-    if (targetCommand.requiresAdmin && !isAdmin(interaction.member as GuildMember)) {
+    const subCommandRequiresAdmin =
+      targetCommand instanceof AggregateCommand && subCommandName
+        ? targetCommand.subCommands.find(s => s.slashCommand.name === subCommandName)?.requiresAdmin
+        : false;
+
+    if (
+      (targetCommand.requiresAdmin || subCommandRequiresAdmin) &&
+      !isAdmin(interaction.member as GuildMember)
+    ) {
       const error = new CommandError('That command requires Admin permissions.');
       await interaction.followUp({ embeds: [buildErrorEmbed(error)] });
```

Comment:
oh also, I think since you're inside a trycatch, you can just `throw new CommandError('That command requires Admin permissions.')` and it'll be turned into an embed anyways